src/api/app/models/workflow.rb

```diff
@@ -74,10 +74,17 @@ def supported_filters
 
   def event_matches_event_filter?
     return true unless supported_filters.key?(:event)
-    return true if filters[:event] == 'push' && scm_webhook.push_event?
-    return true if filters[:event] == 'pull_request' && scm_webhook.pull_request_event?
 
-    false
+    case filters[:events]
```

Comment:
This code is difficult to test, since its method  `event_matches_event_filter?` is private. We wanted to turn this method into a custom validator, but we didn't do it. A service would be a better choice here. It could be easily tested and it would do just one thing, check if the event filter matches the webhook event. I would like to do this in a follow-up PR.

Reply:
> This code is difficult to test, since its method event_matches_event_filter? is private.

Not sure if I understood right, but following the https://github.com/openSUSE/open-build-service/pull/11876#discussion_r752310895, you could just `send` around it, no?

Reply:
In this case, I have a clear alternative to the current state. This will be turned into a service, so I don't need to test the private method with `send`.